src/frontend/js/pages/TeacherTrainingDashboardLoader/index.spec.tsx

```diff
@@ -0,0 +1,103 @@
+import { createMemoryRouter, RouterProvider } from 'react-router-dom';
+import { QueryClientProvider } from '@tanstack/react-query';
+import { render, screen } from '@testing-library/react';
+import fetchMock from 'fetch-mock';
+import { IntlProvider } from 'react-intl';
+
+import { CunninghamProvider } from '@openfun/cunningham-react';
+import { capitalize } from 'lodash-es';
+import {
+  RichieContextFactory as mockRichieContextFactory,
+  UserFactory,
+} from 'utils/test/factories/richie';
+import JoanieSessionProvider from 'contexts/SessionContext/JoanieSessionProvider';
+import { CourseProductRelationFactory } from 'utils/test/factories/joanie';
+import { createTestQueryClient } from 'utils/test/createTestQueryClient';
+import { expectNoSpinner } from 'utils/test/expectSpinner';
+import { TeacherTrainingDashboardLoader } from '.';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockRichieContextFactory({
+    authentication: { backend: 'fonzie', endpoint: 'https://demo.endpoint' },
+    joanie_backend: { endpoint: 'https://joanie.endpoint' },
+  }).one(),
+}));
+
+jest.mock('hooks/useIntersectionObserver', () => ({
+  useIntersectionObserver: (props: any) => {
+    (globalThis as any).__intersection_observer_props__ = props;
+  },
+}));
+
+describe('components/TeacherTrainingDashboardLoader', () => {
+  let nbApiCalls: number;
+  beforeEach(() => {
+    // Joanie providers calls
+    fetchMock.get('https://joanie.endpoint/api/v1.0/orders/', []);
+    fetchMock.get('https://joanie.endpoint/api/v1.0/credit-cards/', []);
+    fetchMock.get('https://joanie.endpoint/api/v1.0/addresses/', []);
+    nbApiCalls = 3;
+  });
+
+  it('do render', async () => {
```

Comment:
renders* ?

Reply:
IMO it would be better to explain a bit more what should be rendered ( at first I had me lost a bit )